pkg/parcacol/sample.go

```diff
@@ -224,100 +225,46 @@ func SeriesToArrowRecord(
 		}
 	}
 
-	return bldr.NewRecord(), nil
-}
-
-// ParquetBufToArrowRecord converts a parquet buffer to an arrow record. If rowsPerRecord is 0, then the entire buffer is converted to a single record.
-func ParquetBufToArrowRecord(ctx context.Context, mem memory.Allocator, buf *dynparquet.Buffer, s *dynparquet.Schema, rowsPerRecord uint) ([]arrow.Record, error) {
-	as, err := pqarrow.ParquetSchemaToArrowSchema(ctx, buf.Schema(), s, logicalplan.IterOptions{})
-	if err != nil {
-		return nil, err
-	}
-
-	bldr := array.NewRecordBuilder(mem, as)
-	defer bldr.Release()
-
-	rows := buf.Rows()
-	defer rows.Close()
-
-	buffSize := 256
-	rowBuf := make([]parquet.Row, buffSize)
-
-	records := []arrow.Record{}
-	recordSize := uint(0)
-	for {
-		n, err := rows.ReadRows(rowBuf)
-		if err != nil && err != io.EOF {
-			return nil, err
+	sortingColumns := []arrowutils.SortingColumn{}
+	arrowFields := as.Fields()
+	for _, col := range schema.SortingColumns() {
```

Comment:
Is there a way to avoid constructing this sorting schema on every record in the happy case? This might not be worth optimizing at this stage especially since this is already likely a huge improvement.